pkg/ctl/install/install.go

```diff
@@ -0,0 +1,421 @@
+	if err != nil {
+		return err
+	}
+	logger.Info("Flux started successfully")
+
+	logger.Info("Committing and pushing Flux manifests to %s", fi.opts.templateParams.GitURL)
+	if err := fi.addCommitAndPushFluxManifests(ctx, cloneDir); err != nil {
+		return err
+	}
+	cleanCloneDir = true
+
+	logger.Info("Flux will operate properly only once it has SSH access to: %s", fi.opts.templateParams.GitURL)
+	logger.Info("please add the following Flux public SSH key to your repository:\n%s", fluxSSHKey.Key)
+	return nil
+}
+
+func (fi *fluxInstaller) cloneRepo(ctx context.Context) (string, error) {
+	cloneDir, err := ioutil.TempDir(os.TempDir(), "eksctl-install-flux-clone")
+	if err != nil {
+		return "", fmt.Errorf("cannot create temporary directory: %s", err)
+	}
+	cloneCtx, cloneCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
+	defer cloneCtxCancel()
+	args := []string{"clone", "-b", fi.opts.templateParams.GitBranch, fi.opts.templateParams.GitURL, cloneDir}
+	err = runGitCmd(cloneCtx, cloneDir, args...)
+	return cloneDir, err
+}
+
+func (fi *fluxInstaller) addCommitAndPushFluxManifests(ctx context.Context, cloneDir string) error {
+	// git add
+	addCtx, addCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
+	defer addCtxCancel()
+	if err := runGitCmd(addCtx, cloneDir, "add", "--", fi.opts.gitFluxPath); err != nil {
+		return err
+	}
+
+	// Confirm there is something to commit, otherwise move on
+	diffCtx, diffCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
+	defer diffCtxCancel()
+	if err := runGitCmd(diffCtx, cloneDir, "diff", "--cached", "--quiet", "--", fi.opts.gitFluxPath); err == nil {
+		logger.Info("Nothing to commit (the repository contained identical manifests), moving on")
+		return nil
+	} else if _, ok := err.(*exec.ExitError); !ok {
+		return err
+	}
+
+	// git commit
+	commitCtx, commitCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
+	defer commitCtxCancel()
+	args := []string{"commit",
+		"-m", "Add Initial Flux configuration",
+		fmt.Sprintf("--author=%s <%s>", fi.opts.templateParams.GitUser, fi.opts.templateParams.GitEmail),
+	}
+	if err := runGitCmd(commitCtx, cloneDir, args...); err != nil {
+		return err
+	}
+
+	// git push
+	pushCtx, pushCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
+	defer pushCtxCancel()
+	return runGitCmd(pushCtx, cloneDir, "push")
+}
+
+func (fi *fluxInstaller) applyManifests(manifestsMap map[string][]byte) error {
+	// TODO: initialise the client elsewhere so that a mock client can easily be dependency-injected for testing purposes.
+	client, err := kubernetes.NewRawClient(fi.k8sClientSet, fi.k8sRestConfig)
+	if err != nil {
+		return err
+	}
+
+	manifests := kubernetes.ConcatManifestValues(manifestsMap)
+	objects, err := kubernetes.NewRawExtensions(manifests)
+	if err != nil {
+		return err
+	}
+	for _, object := range objects {
+		resource, err := client.NewRawResource(object)
+		if err != nil {
+			return err
+		}
+		status, err := resource.CreateOrReplace(false)
```

Comment:
There is actually an issue with using `CreateOrReplace`: now the "amend" mode will work in a rather odd way.
Indeed, if the end-user was to amend manifests, `eksctl` now:

- wouldn't take these modifications into account, 
- would apply whatever it has in memory,
- would git-commit-push the modifications,
- later, Flux would pick the modifications up and apply them.

I believe this may lead to confusion.
I'd be tempted to either drop the "amend" mode (was this a strong requirement?), or re-read the manifests from files, or go back to `kubectl`, whatever we collectively prefer. WDYT @2opremio, @errordeveloper?

Reply:
For now, I'll re-read the manifests, as I think this is the best option.